### PR TITLE
feat: Add java client MultiJoin

### LINKED ISSUE #5883: Java client MultiJoin
Part of #5858

### PR DESCRIPTION
**Pull request by devinrsmith**
Fixes #5883

### CODE REVIEW

**Comment by devinrsmith** [comment]
Still need to add some tests, but should otherwise work.

File: qst/src/main/java/io/deephaven/qst/TableCreator.java [hunk 1]
```diff
@@ -7,6 +7,8 @@
 import io.deephaven.qst.column.Column;
 import io.deephaven.qst.table.EmptyTable;
 import io.deephaven.qst.table.InputTable;
+import io.deephaven.qst.table.MultiJoinInput;
+import io.deephaven.qst.table.MultiJoinTable;
```
Comment (rcaudy) [inline qst/src/main/java/io/deephaven/qst/TableCreator.java:11]:
```suggestion
```

File: qst/src/main/java/io/deephaven/qst/table/StackTraceMixInCreator.java [hunk 2]
```diff
@@ -6,11 +6,14 @@
 import io.deephaven.api.TableOperations;
 import io.deephaven.qst.TableCreator;
 
+import java.util.ArrayList;
```
Comment (rcaudy) [inline qst/src/main/java/io/deephaven/qst/table/StackTraceMixInCreator.java:9]:
```suggestion
```

File: qst/src/main/java/io/deephaven/qst/table/ParentsVisitor.java [hunk 3]
```diff
@@ -302,6 +302,11 @@ public Stream<TableSpec> visit(DropColumnsTable dropColumnsTable) {
         return single(dropColumnsTable);
     }
 
+    @Override
```
Comment (rcaudy) [inline qst/src/main/java/io/deephaven/qst/table/ParentsVisitor.java:305]: Delete the `HashSet` import.

File: engine/table/src/main/java/io/deephaven/engine/table/impl/TableCreatorImpl.java [hunk 4]
```diff
@@ -85,6 +88,14 @@ public final Table of(InputTable inputTable) {
         return InputTableAdapter.of(inputTable);
     }
 
+    @Override
+    public final Table multiJoin(List<io.deephaven.qst.table.MultiJoinInput<Table>> multiJoinInputs) {
+        return MultiJoinFactory.of(multiJoinInputs.stream().map(this::adapt).toArray(MultiJoinInput[]::new)).table();
+    }
+
+    private MultiJoinInput adapt(io.deephaven.qst.table.MultiJoinInput<Table> input) {
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/TableCreatorImpl.java:96]: Static?

File: java-client/session/src/main/java/io/deephaven/client/impl/TableHandleManagerBase.java [hunk 5]
```diff
@@ -49,6 +53,19 @@ public final TableHandle of(InputTable inputTable) {
         return handle(inputTable);
     }
 
+    @Override
+    public final TableHandle multiJoin(List<MultiJoinInput<TableHandle>> multiJoinInputs) {
+        MultiJoinTable.Builder builder = MultiJoinTable.builder();
+        for (MultiJoinInput<TableHandle> input : multiJoinInputs) {
+            builder.addInputs(MultiJoinInput.<TableSpec>builder()
```
Comment (rcaudy) [inline java-client/session/src/main/java/io/deephaven/client/impl/TableHandleManagerBase.java:60]:
```suggestion
            //noinspection resource We're not making new TableHandles here
            builder.addInputs(MultiJoinInput.<TableSpec>builder()
```

File: java-client/session-dagger/src/test/java/io/deephaven/client/MultiJoinTest.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import io.deephaven.qst.table.MultiJoinTable;
+import io.deephaven.qst.table.TableSpec;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class MultiJoinTest extends DeephavenSessionTestBase {
+
+    @Test
+    public void muliJoinTableExecute() throws TableHandleException, InterruptedException {
```
Comment (rcaudy) [inline java-client/session-dagger/src/test/java/io/deephaven/client/MultiJoinTest.java:22]:
```suggestion
    public void multiJoinTableExecute() throws TableHandleException, InterruptedException {
```

**Review by rcaudy — APPROVED** [review]

**Pull request merged**